javaslang/src/main/java/javaslang/collection/Queue.java

```diff
@@ -595,34 +530,34 @@ public T peek() {
 
     @Override
     public Queue<Queue<T>> combinations() {
-        return toList().combinations().map(Queue::ofAll).toQueue();
+        return ofAll(toList().combinations().map(Queue::ofAll));
```

Comment:
we have multiple `Queue` implementations now, maybe we can extract common functionality this way somehow